src/helpers/wallet-precalculation.helper.js

```diff
@@ -0,0 +1,234 @@
+    'sample garment fun depart various renew require surge service undo cinnamon squeeze hundred nasty gasp ridge surge defense relax turtle wet antique october occur',
+    'intact wool rigid diary mountain issue tiny ugly swing rib alone base fold satoshi drift poverty autumn mansion state globe plug ancient pudding hope',
+    'monster opinion bracket aspect mask labor obvious hat matrix exact canoe race shift episode plastic debris dash sort motion juice leg mushroom maximum evidence',
+    'tilt lab swear uncle prize favorite river myth assault transfer venue soap lady someone marine reject fork brain swallow notice glad salt sudden pottery',
+  ],
+  pubkeys: [
+    'xpub6CvvCBtHqFfErbcW2Rv28TmZ3MqcFuWQVKGg8xDzLeAwEAHRz9LBTgSFSj7B99scSvZGbq6TxAyyATA9b6cnwsgduNs9NGKQJnEQr3PYtwK',
+    'xpub6CA16g2qPwukWAWBMdJKU3p2fQEEi831W3WAs2nesuCzPhbrG29aJsoRDSEDT4Ac3smqSk51uuv6oujU3MAAL3d1Nm87q9GDwE3HRGQLjdP',
+    'xpub6BwNT613Vzy7ARVHDEpoX23SMBEZQMJXdqTWYjQKvJZJVDBjEemU38exJEhc6qbFVc4MmarN68gUKHkyZ3NEgXXCbWtoXXGouHpwMEcXJLf',
+    'xpub6DCyPHg4AwXsdiMh7QSTHR7afmNVwZKHBBMFUiy5aCYQNaWp68ceQXYXCGQr5fZyLAe5hiJDdXrq6w3AXzvVmjFX9F7EdM87repxJEhsmjL',
+    'xpub6CgPUcCCJ9pAK7Rj52hwkxTutSRv91Fq74Hx1SjN62eg6Mp3S3YCJFPChPaDjpp9jCbCZHibBgdKnfNdq6hE9umyjyZKUCySBNF7wkoG4uK',
+  ],
+  walletConfig: {
+    pubkeys: [],
+    total: 5,
+    minSignatures: 3,
+  }
+};
+multisigWalletsData.walletConfig.pubkeys = multisigWalletsData.pubkeys;
+
+export class WalletPrecalculationHelper {
+  WALLETS_FILENAME = '';
+
+  walletsDb = [];
+
+  /**
+   * Initializes the helper with a filename to sync the local wallet storage with.
+   * @param walletsFilename
+   */
+  constructor(walletsFilename) {
+    this.WALLETS_FILENAME = walletsFilename || './tmp/wallets.txt';
+  }
+
+  /**
+   * Generates 22 addresses for a wallet 24-word seed.
+   * @param [params]
+   * @param {string} [params.words] Optional wallet seed words. If empty, generates a new wallet
+   * @param {number} [params.addressIntervalStart=0] Optional interval start index
+   * @param {number} [params.addressIntervalEnd=22] Optional interval end index
+   * @param {{minSignatures:number,wordsArray:string[]}} [params.multisig] Optional multisig object
+   * @returns {{addresses: string[], words: string}}
+   */
+  static generateWallet(params = {}) {
+    const timeStart = Date.now().valueOf();
+    const walletCfg = {
+      seed: params.words || '',
+      connection: {},
+      password: '123',
+      pinCode: '123',
+      multisig: false
+    };
+
+    walletCfg.connection = new Connection({
+      network: config.network,
+      servers: [config.server],
+      connectionTimeout: config.connectionTimeout,
+    });
+
+    let multisigDebugData = null;
+    if (params.multisig) {
+      const pubkeys = params.multisig.wordsArray.map(w => walletUtils.getMultiSigXPubFromWords(w));
```

Comment:
`params.multisig` is not on the docstring

Reply:
It's on line 56, but it has a different syntax since it implements a full object type description. This was chosen because it's only used here, a full `@typedef` would be too much.
```js
* @param {{minSignatures:number,wordsArray:string[]}} [params.multisig] Optional multisig object
```